lib/datasource/docker/index.spec.ts

```diff
@@ -263,6 +263,46 @@ describe('datasource/docker/index', () => {
       });
     });
 
+    it('passes session token to ECR client', async () => {
+      httpMock
+        .scope(amazonUrl)
+        .get('/')
+        .reply(401, '', {
+          'www-authenticate': 'Basic realm="My Private Docker Registry Server"',
+        })
+        .head('/node/manifests/some-tag', undefined, {
+          reqheaders: { authorization: 'Basic abc' },
```

Comment:
This looks wrong to me, shouldn't this be `Bearer test_token` ?

Reply:
I copied this from the test above.

`Basic` is indeed correct, according to https://docs.aws.amazon.com/AmazonECR/latest/userguide/registry_auth.html#registry_auth_http.

I don't understand what the `abc` is about. No matter to what value I set this, the test stays green.

Reply:
Ok, but I'm sure this should have the ecr token.

Strange, i need to test that locally but won't like to block this PR.

Can you open a refactor issue for this: validate docker auth headers in tests

Reply:
Yes, I agree! Will try to fix later tonight and split the refactor out if necessary.

Reply:
Seems that `reqheaders` does not match the header values. The request is matched no matter which header value is configured.

With https://github.com/renovatebot/renovate/pull/11967/commits/676fd9f0798ce247d0c14e2c3f98c5e473915e5a I switched to `matchHeaders` which behaves properly. That also made the test above fail so that I had to update the value to `test_token` instead of `abc`.

Reply:
FWIW there is a pretty old Issue in the the nock repo which describes the same behavior: https://github.com/nock/nock/issues/748